crates/storage/src/iter.rs

```diff
@@ -340,6 +395,228 @@ pub trait IteratorOverTable {
 
 impl<S> IteratorOverTable for S {}
 
+/// A trait for iterating over writes to a `Mappable` table.
+pub trait IterableTableWrites<M>
+where
+    M: Mappable,
+{
+    /// Returns an iterator over the all keys in the table with a prefix after a specific start key.
+    fn iter_table_writes_keys<P>(
+        &self,
+        prefix: Option<P>,
+        start: Option<&M::Key>,
+        direction: Option<IterDirection>,
+    ) -> BoxedIter<super::Result<M::OwnedKey>>
+    where
+        P: AsRef<[u8]>;
+
+    /// Returns an iterator over the all entries in the table with a prefix after a specific start key.
+    #[allow(clippy::type_complexity)]
+    fn iter_table_writes<P>(
+        &self,
+        prefix: Option<P>,
+        start: Option<&M::Key>,
+        direction: Option<IterDirection>,
+    ) -> BoxedIter<super::Result<(M::OwnedKey, Option<M::OwnedValue>)>>
+    where
+        P: AsRef<[u8]>;
+}
+
+impl<Column, M, S> IterableTableWrites<M> for S
+where
+    M: TableWithBlueprint<Column = Column>,
+    M::Blueprint: BlueprintInspect<M, S>,
+    S: IterableStoreWrites<Column = Column>,
+{
+    fn iter_table_writes_keys<P>(
+        &self,
+        prefix: Option<P>,
+        start: Option<&M::Key>,
+        direction: Option<IterDirection>,
+    ) -> BoxedIter<crate::Result<M::OwnedKey>>
+    where
+        P: AsRef<[u8]>,
+    {
+        #[allow(clippy::redundant_closure)]
+        // false positive: https://github.com/rust-lang/rust-clippy/issues/14215
+        let encoder = start
+            .map(|start| <M::Blueprint as BlueprintCodec<M>>::KeyCodec::encode(start));
+
+        let start = encoder.as_ref().map(|encoder| encoder.as_bytes());
+
+        IterableStoreWrites::iter_store_write_keys(
+            self,
+            M::column(),
+            prefix.as_ref().map(|p| p.as_ref()),
+            start.as_ref().map(|cow| cow.as_ref()),
+            direction.unwrap_or_default(),
+        )
+        .map(|res| {
+            res.and_then(|key| {
+                let key =
+                    <M::Blueprint as BlueprintCodec<M>>::KeyCodec::decode(key.as_slice())
+                        .map_err(|e| crate::Error::Codec(anyhow::anyhow!(e)))?;
+                Ok(key)
+            })
+        })
+        .into_boxed()
+    }
+
+    fn iter_table_writes<P>(
+        &self,
+        prefix: Option<P>,
+        start: Option<&M::Key>,
+        direction: Option<IterDirection>,
+    ) -> BoxedIter<super::Result<(M::OwnedKey, Option<M::OwnedValue>)>>
+    where
+        P: AsRef<[u8]>,
+    {
+        #[allow(clippy::redundant_closure)]
+        // false positive: https://github.com/rust-lang/rust-clippy/issues/14215
+        let encoder = start
+            .map(|start| <M::Blueprint as BlueprintCodec<M>>::KeyCodec::encode(start));
```

Comment:
seeing as you are using this pattern a lot, would it be meaningful to add -

```rs
pub type KeyCodecOf<M, C> = <<M as TableWithBlueprint<Column = C>> as BlueprintCodec<M>::KeyCodec;
```

and then in this code:
```suggestion
            .map(|start| KeyCodecOf::<M, Column>::encode(start));
```

and elsewhere ? im ok with the current implementation, it has less indirection while reviewing the code

Reply:
same for decode, of course

Reply:
I don't feel like this is used enough to justify the alias.